src/EnergyPlus/DXCoils.cc

```diff
@@ -17168,6 +17168,16 @@ void CalcVRFHeatingCoil_FluidTCtrl(EnergyPlusData &state,
         (PartLoadRatio > 0.0) && (OutdoorDryBulb > thisDXCoil.MinOATCompressor)) {
 
         TotCap = thisDXCoil.RatedTotCap(Mode);
+        HeatingCapacityMultiplier = 1.0;
+        // Modify total heating capacity based on defrost heating capacity multiplier
+        // MaxHeatCap passed from parent object VRF Condenser and is used to limit capacity of TU's to that available from condenser
+        if (present(MaxHeatCap)) {
+            TotCapAdj = min(MaxHeatCap, TotCap * HeatingCapacityMultiplier);
+            TotCap = min(MaxHeatCap, TotCap);
+        } else {
+            TotCapAdj = TotCap * HeatingCapacityMultiplier;
+        }
+
```

Comment:
There are only 2 calls to CalcVRFHeatingCoil_FluidTCtrl. Is there a reason why MaxHeatCap can't be passed in both calls?

Reply:
The one called in InitDXCoil does not pass in a MaxHeatCap. I inserted a breakpoint there and saw that at that moment the output from `LimitTUCapacity`, `state.dataHVACVarRefFlow->MaxHeatingCapacity(1)`, was 0. It has not been properly updated yet. It would be bad to pass in 0 as the capacity limit. The `CalcVRFHeatingCoil_FluidTCtrl` inside `InitDXCoil` is triggered only once, so it won't ever have a non-zero `MaxHeatCap` input. Maybe that's why it's not passed in? A few other coil types also did the same thing.
<img width="1474" alt="image" src="https://github.com/NREL/EnergyPlus/assets/13421161/c02c9203-dc66-49f0-8245-5e4d3d11569e">

Reply:
See the use of this variable:

    constexpr Real64 MaxCap(1.0e+20); // limit of zone terminal unit capacity

Reply:
Or at the initialization step, would there already be a more sensible max capacity? rather than a 1.0e+20?

Reply:
For initialization the TU coil after sizing, the TU does not need to be limited. The call with no max cap argument only happens 1 time. You could pass MaxCap there without any reprocussions. And then that argument would no longer need to be optional. Just a thought.

Reply:
What I mean is, instead of doing this:

        // MaxHeatCap passed from parent object VRF Condenser and is used to limit capacity of TU's to that available from condenser
        if (present(MaxHeatCap)) {
            TotCapAdj = min(MaxHeatCap, TotCap * HeatingCapacityMultiplier);
            TotCap = min(MaxHeatCap, TotCap);
        } else {
            TotCapAdj = TotCap * HeatingCapacityMultiplier;
        }

Just pass in MaxCap and you can reduce this code to:

            TotCapAdj = min(MaxHeatCap, TotCap * HeatingCapacityMultiplier);
            TotCap = min(MaxHeatCap, TotCap);

Reply:
Or better yet, this:

        TotCap = min(MaxHeatCap, TotCap);

        // Calculate full load outlet conditions
        FullLoadOutAirEnth = InletAirEnthalpy + (TotCap  * HeatingCapacityMultiplier) / AirMassFlow;

Although, I am not sure why TotCapAdj is only use here? Does total capacity include defrost or doesn't it?

Reply:
Thanks for the clarification. Now I understand what you meant. Indeed if the MaxHeatCap argument is not optional and I pass in the `MaxCap` (a constant of value 1.0e+20), it won't need to be an optional argument. Is an optional argument preferred or do we try to avoid it? If we don't want the optional argument, should we pass in something like `numeric_limits<float>::infinity()` to be more explicit? Maybe @Myoldmopar and @amirroth could also chime in on this?

For the other question, it seems `TotCap` doesn't include defrost. It is computed like this
`TotCap = thisDXCoil.RatedTotCap(Mode) * TotCapFlowModFac * TotCapTempModFac;`